### PR TITLE
Case 21264: HMD Login Encouragement Screen is not Usable

### PR DESCRIPTION
**Pull request by danteruiz**
Ticket - https://highfidelity.manuscript.com/f/cases/21264/HMD-Login-Encouragement-Screen-is-not-Usable

### CODE REVIEW

**Comment by danteruiz** [comment]
Test plan - login screen on start up should work

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17661/HighFidelity-Beta-PR15039-93601ff.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17661/HighFidelity-Quest-Beta-PR15039-93601ff.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17661/HighFidelity-Beta-PR15039-93601ff.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17661/HighFidelity-Beta-Interface-PR15039-93601ff.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17661/HighFidelity-Beta-PR15039-93601ff.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17661/HighFidelity-Beta-Interface-PR15039-93601ff.exe

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -4966,6 +4966,18 @@ void Application::idle() {
         }
     }
 
+    {
+        if (!_keyboardFocusedEntity.get().isInvalidID() && !_isProxyWindowSet) {
```
Comment (SamGondelman) [inline interface/src/Application.cpp]: discussed alternative with dante

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17670/HighFidelity-Beta-PR15039-6d8fca0.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17670/HighFidelity-Quest-Beta-PR15039-6d8fca0.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17670/HighFidelity-Beta-PR15039-6d8fca0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17670/HighFidelity-Beta-Interface-PR15039-6d8fca0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17670/HighFidelity-Beta-PR15039-6d8fca0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17670/HighFidelity-Beta-Interface-PR15039-6d8fca0.exe

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -5811,6 +5820,10 @@ void Application::setKeyboardFocusEntity(const QUuid& id) {
             if (properties.getVisible()) {
                 auto entities = getEntities();
                 auto entityId = _keyboardFocusedEntity.get();
+                auto entityItemRenderable = entities->renderableForEntityId(entityId);
+                if (!entityItemRenderable) {
+                    _keyboardFocusWaitingOnRenderable = true;
+                }
                 if (entities->wantsKeyboardFocus(entityId)) {
```
Comment (SamGondelman) [inline interface/src/Application.cpp]: `} else if (entities->wantsKeyboardFocus(entityId)) {`

File: interface/src/Application.cpp [hunk 3]
```diff
@@ -4966,6 +4966,14 @@ void Application::idle() {
         }
     }
 
+    {
+        if (_keyboardFocusWaitingOnRenderable) {
```
Comment (SamGondelman) [inline interface/src/Application.cpp]:
```
if (_keyboardFocusWaitingOnRenderable && getEntities()->renderableForEntityId(entityId)) {
     _keyboardFocusWaitingOnRenderable = false;
```

File: interface/src/Application.cpp [hunk 4]
```diff
@@ -5794,13 +5802,14 @@ void Application::setKeyboardFocusEntity(const QUuid& id) {
...
                 emit loginDialogFocusEnabled();
-            } else {
+            } else if (!_keyboardFocusWaitingOnRenderable) {
                 // that's the only entity we want in focus;
                 return;
             }
         }
 
         _keyboardFocusedEntity.set(id);
+        _keyboardFocusWaitingOnRenderable = false;
```
Comment (SamGondelman) [inline interface/src/Application.cpp]: remove

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17680/HighFidelity-Beta-PR15039-edac56e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17680/HighFidelity-Beta-Interface-PR15039-edac56e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17680/HighFidelity-Beta-PR15039-edac56e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17680/HighFidelity-Quest-Beta-PR15039-edac56e.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17680/HighFidelity-Beta-PR15039-edac56e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17680/HighFidelity-Beta-Interface-PR15039-edac56e.exe

**Comment by qlimguicheteau** [comment]
Pre-Merge Testing

**Comment by qlimguicheteau** [comment]
PASS
Oculus works
Vive works

**Pull request merged**